### ISSUE TITLE
Delayed Notifications when the app is in the background

### ISSUE THREAD

**Issue by msdosfx** [mollyim/mollyim-android#210]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

Molly Foss, notifications became unreliable as of a month ago or so. Notifications won't interrupt if I am listening to music or a podcast. Most of the time I get no notification until I actually open the app. I receive no notification or ringing for an incoming call, and with an outgoing voice call the other person can't answer. I have no idea whether this is a Graphene OS issue or it is a Molly issue. I only know that things were working fine for a while and then they weren't. I recently reenabled the sandboxed google play services, but the notification issue was present before then, and my understanding was that the Molly Foss version didn't use and google push notifications regardless of Play being enabled. Notifications are enabled in the app. 

### Steps to reproduce

1. receive a text message on Molly from a Signal user
2.I get no notification until I actually open the app.
3. I expect to get a notification within minutes of the message being sent. Definitely within hours, but I don't unless I open the app. It also doesn't do notifications over running apps such as Poweramp, Symphonium, or Antennapod. I expect it to ring or provide a notification sound if someone calls or texts me from Signal. 

### Molly version

v6.30.4-1-FOSS

### Android version

Graphene OS Android 12

### Device

Pixel 4a

### Link to debug log

https://debuglogs.org/android/6.30.4/6298345bd914630b7d8dfa3eafc44e20ca8289f1d214532c7466aac904d6f3de

**Comment by hiddenimages** [comment]
I have this same issue with foss version.   I just installed Molly on Sep 8/23 after just hearing about it for the first time this week.   Exactly the same scenario...  I only get a notification after opening the app.    Pixel 7 Pro; Graphene OS.  Google services not installed/

**Comment by valldrac** [comment]
@msdosfx I've checked the log, and I don't see any issues with notifications. However, I did notice that the app couldn't connect to the server for quite some time while in background, so it seems like a connectivity problem or similar. Are you using a VPN of application firewall?

**Comment by msdosfx** [comment]
@valldrac I use an always-on VPN and I use the system level Android permissions that are available on Graphene OS. I have since reverted back to Signal with GPservices to see if this issue persists, so I can't tell you exactly which permissions were active on Molly, but I definitely made sure that the Molly app had notifications permissions and network permissions. Phone, Camera, and Photos were also granted.  Another thing to note is that I pretty much keep my phone in airplane mode with wifi on all of the time. So if I am away from wifi, my phone won't be able to contact the servers. I tested and duplicated the failed notification while I definitely had a wifi connection though a few times. I would try to do some testing to chase this bug, but b/c of the nature of Signal/Molly being only on one mobile device I can't really do that easily without creating communication problems for myself. Thank you for checking my logs.

**Comment by valldrac** [comment]
I see. Thanks for helping me understand the issue. Next time, please make sure to note the time the message is sent and the delay in the notification so I can match it up with the log entries.

**Comment by c0mmando** [comment]
I have the same issue, also GrapheneOS, started about a month ago. It appears the background connection to signal servers is failing. Sometimes the background connection works, and sometimes I won't get notifications until I open the app. 

This is such a serious issue that I may need to revert to using Signal over Molly.

**Comment by msdosfx** [comment]
> I have the same issue, also GrapheneOS, started about a month ago. It appears the background connection to signal servers is failing. Sometimes the background connection works, and sometimes I won't get notifications until I open the app.
> 
> This is such a serious issue that I may need to revert to using Signal over Molly.
> 
> Issue persists as of v6.40.4-1-FOSS

If you can, next time it happens, send your logs to the developers here so they can track it down. I had to switch back to Signal b/c I was missing too many messages, before I could really help them troubleshoot the issue

**Comment by aroddick** [comment]
Just want to add another data point. This has started happening to me as well and only after I started keeping my phone in airplane mode with wifi. When I reconnect to wifi I don't get notifications until I open the app. I am using an always on VPN as well. Will try to catch logs of it next time it happens.

**Comment by valldrac** [comment]
I keep asking for the debug log, just include the timestamp of when the message is sent and the notification delay.

**Comment by valldrac** [comment]
We've released several new versions since this issue was reported. To keep things tidy, we're closing this for now. If you still find this issue again or have any new information, please feel free to open a new ticket.
